macros/models/_project/yield_yak/yield_yak_reinvests.sql

```diff
@@ -0,0 +1,138 @@
+{%- macro yield_yak_reinvests(
+        blockchain = null
+    )
+-%}
+
+{%- set namespace_blockchain = 'yield_yak_' + blockchain -%}
+
+WITH
+
+{% if is_incremental() -%}
+existing_contracts AS (
+    SELECT
+        t.contract_address
+        , MAX(t.block_number) AS latest_block_number
+        , MAX_BY(t.ratio, (t.block_number, t.tx_index, t.evt_index)) AS latest_ratio
+        , MAX_BY(t.block_time, (t.block_number, t.tx_index, t.evt_index)) AS latest_block_time
+        , MAX_BY(t.recent_reinvest_info, (t.block_number, t.tx_index, t.evt_index)) AS latest_recent_reinvest_info
+    FROM
+    {{ this }} t
+    GROUP BY t.contract_address
+),
+{% endif -%}
+
+combined AS (
+    {%- for strategy in yield_yak_strategies(blockchain) %}
+        SELECT
+            s.contract_address
+            , s.evt_tx_hash AS tx_hash
+            , s.evt_index
+            , s.evt_block_time AS block_time
+            , s.evt_block_number AS block_number
+            , s.newTotalDeposits AS new_total_deposits
+            , s.newTotalSupply AS new_total_supply
+            , CASE WHEN s.newTotalSupply = 0 THEN 1 ELSE (1.0 * s.newTotalDeposits) / (1.0 * s.newTotalSupply) END AS ratio
+            {%- if is_incremental() %}
+            , c.latest_ratio
+            , c.latest_block_time
+            , c.latest_recent_reinvest_info
+            {%- endif %}
+        FROM {{ source(namespace_blockchain, strategy + '_evt_Reinvest') }} s
+        {%- if is_incremental() %}
+        LEFT JOIN
+        existing_contracts c
+            ON c.contract_address = s.contract_address
+        WHERE
+            ({{ incremental_predicate('s.evt_block_time') }}
+            AND s.evt_block_number > c.latest_block_number)
+            OR c.contract_address IS NULL -- This line allows for new contract_addresses being appended that were not already included in previous runs but also allows their entire historical data to be loaded
+        {%- endif %}
+        {% if not loop.last -%}
+        UNION ALL
+        {%- endif -%}
+    {%- endfor %}
+),
+
+add_tx_index AS (
+    SELECT
+        c.*
+        , t.index AS tx_index
+        , t."from" AS reinvest_by_address
+    FROM combined c
+    LEFT JOIN
+    {{ source(blockchain, 'transactions') }} t
+        ON t.hash = c.tx_hash
```

Comment:
same idea on joins here from deposits/withdraws macro

Reply:
Yep, can enhance the joins (hadn't realise that would improve performance - good to know!). And yes I think it can be an inner join (there shouldn't be anything present in the decoded tables which does not have a corresponding row in the `transactions` tables). There's no `block_date` in the decoded tables so will do the `block_date` join condition using `date_trunc` as you've written.

Reply:
Am taking the approach of adding `block_date` as a column for the deposits/withdraws/reinvests models also (think it might be handy for daily time series data anyway).